src/helix.idea.vim

```diff
@@ -0,0 +1,198 @@
+"""""""""""""""""""""""""
+" IdeaVim support       "
+"""""""""""""""""""""""""
+set nohlsearch                    " no highlight search results
+set number                        " line numbers
+set showmode                      " show current mode
+set gdefault                      " substitute all occurrences in line per default
+set history=10000                 " keep x lines of command line history
+set scrolloff=5 sidescrolloff=10  " keep some lines before and after the cursor visible
+set visualbell                    " no beep
+set smartcase                     " no ignore case when pattern is uppercase
+set incsearch                     " show search results while typing
+set wrapscan                      " searches wrap around the end of the file
+
+set ideajoin
+set idearefactormode=keep
+
+" Plugins
+set commentary
+set highlightedyank
+set argtextobj
+set matchit
+set multiple-cursors
+
+nnoremap , <Esc><Esc>
+xnoremap , <Esc><Esc>
+vnoremap , <Esc><Esc>
+
+xnoremap <Esc>o o
+nnoremap o o
+
+nnoremap d vd
+xnoremap d dv
+
+" Movement
+unmap <Left>
+unmap <Right>
+unmap <Down>
+unmap <Up>
+
+noremap t vt
+noremap T vT
+noremap f vf
+noremap F vF
+
+" Select/extend
+nnoremap e ve
+nnoremap E vE
+nnoremap w vw
+nnoremap W vW
+nnoremap b vb
+nnoremap B vB
+
+noremap X 0V
+xnoremap X <nop>
+
+" Match mode
+noremap ma) <Esc>va(
+noremap ma} <Esc>va{
+noremap ma> <Esc>va<
+noremap ma[ <Esc>va[
+
+noremap mi) <Esc>vi)
+noremap mi{ <Esc>vi}
+noremap mi< <Esc>vi>
+noremap mi[ <Esc>vi]
+
+" Goto mode
+nmap gr <Action>(FindUsages)
+nmap gi <Action>(GotoImplementation)
+nmap gd <Action>(GotoDeclaration)
+nmap gT <Action>(GotoTest)
+nmap gp <Action>(PreviousTab)
+nmap gy <Action>(GotoTypeDeclaration)
+
+" Requires plugin: KJump
+nmap gw <Action>(KJumpAction.Word0)
+xmap gw <Action>(KJumpAction.Word0)
+
+" Requires plugin: String manipulation
+xmap & <Action>(StringManipulation.AlignSelections)
+
+" Selection manipulation
+nmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+vmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+nmap <A-o> <Action>(EditorSelectWord)
+xmap <A-o> <Action>(EditorSelectWord)
+nmap <A-up> <Action>(EditorSelectWord)
+xmap <A-up> <Action>(EditorSelectWord)
+nmap <A-i> <Action>(EditorUnSelectWord)
+xmap <A-i> <Action>(EditorUnSelectWord)
+nmap <A-down> <Action>(EditorUnSelectWord)
+xmap <A-down> <Action>(EditorUnSelectWord)
+nmap C <Action>(EditorCloneCaretBelow)
+xmap C <Action>(EditorCloneCaretAbove)
+
+xnoremap ~ Ugv
+nnoremap ~ vU
+xnoremap ` ugv
+nnoremap ` vu
+
+nnoremap * v"ay:<C-u>let @/='<C-r>a'<CR>
+xnoremap * "ay:<C-u>let @/='<C-r>a'<CR>
+
+" View mode
+nmap zc <Action>(EditorScrollToCenter)
```

Comment:
Is there a reason nmap is used instead of nnoremap? In general nnoremap should be the default, and nmap is used when you want a recursive binding.

Reply:
Thanks! I was under the impression `*noremap` means that the right hand side of the mapping will be handled with Vim defaults. I also saw many IdeaVim dotfiles use `map` and `nmap` all over the config.

I assume this could then break some mappings defined in `helix.vim`? :thinking:

Reply:
In this case it probably makes no difference, since you're not actually referencing keys on the right hand side. But in general it's a good idea to use noremap by default. This way exceptions to the rule clearly stand out, which makes trouble shooting easier (in my experience, but it's also considered best practice in vim-land).

Reply:
It appears `noremap` mappings require using `:action FooBar<CR>` rather than the newer format `<Action>(Foobar)`. 

[IdeaVim docs](https://github.com/JetBrains/ideavim?tab=readme-ov-file#executing-actions) say this:

> Please don't use :action in mappings. Use <Action> instead.

No clue why that is, but perhaps the emulator is able to handle the actins more reliably/efficiently when given in this format. It's probably a safer bet to do that?